src/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs

```diff
@@ -394,7 +394,7 @@ internal string GetConnectAttrString(ODBC32.SQL_ATTR attribute)
                 }
                 if ((ODBC32.RetCode.SUCCESS == retcode) || (ODBC32.RetCode.SUCCESS_WITH_INFO == retcode))
                 {
-                    value = Encoding.Unicode.GetString(buffer, 0, Math.Min(cbActual, buffer.Length));
+                    value = (BitConverter.IsLittleEndian ? Encoding.Unicode : Encoding.BigEndianUnicode).GetString(buffer, 0, Math.Min(cbActual, buffer.Length));
```

Comment:
Are strings the only Types which are not working right, on a BigEndian architecture?

Reply:
Endianness should be handled properly by the system P/Invoke marshal, except cases like this were data is being explicitly handled from byte arrays, and using the improper Encoding object.